airflow/www/blueprints.py

```diff
@@ -32,6 +37,26 @@ def index():
 
 @routes.route('/health')
 def health():
-    """ We can add an array of tests here to check the server's health """
-    content = Markup(markdown.markdown("The server is healthy!"))
-    return content
+    session = settings.Session()
+    DM = models.DagModel
+    payload = {}
+
+    latest_scheduler_run = None
+    payload['metadatabase'] = {'status': 'healthy'}
+    try:
+        latest_scheduler_run = session.query(func.max(DM.last_scheduler_run)).scalar()
+    except Exception as _:
+        payload['metadatabase']['status'] = 'unhealthy'
+
+    if not latest_scheduler_run:
+        scheduler_status = 'unhealthy'
+    else:
+        if timezone.utcnow() - latest_scheduler_run <= timedelta(seconds=30):
```

Comment:
could you comment on this 30 seconds magic number and why we pick 30 seconds to define the health/non-health for scheduler?

Reply:
I have put this threshold into `airflow.cfg` as a configuration item, with default value 30 seconds.